### PR TITLE
Support given for enforcing DirectInvocationOnMock :  issue 3396

### PR DESCRIPTION
**Pull request by RobertTheBlair**
Support given for enforcing DirectInvocationOnMock :  issue 3396

* Mockito supports BDDMockito, using "given" to correctly define mock behavior, instead of plain Mockito "when"
* test some use cases that parallel existing cases

---
This is my first PR for this Repo, so I'm approaching with all humbleness.

My team at work saw this error prone warning show up, and 95% of the occurrences were from our use of BDD style `given` calls.   Rather than ignoring the warning outright, I'd like to fix it here.

### CODE REVIEW

**Comment by RobertTheBlair** [comment]
The missing CLA is from me using my work email in the commit - will fix it shortly, commit using personal email address

**Comment by RobertTheBlair** [comment]
Still looking to get this reviewed & approved - everything should be ready, except the build does not pass as I've not been approved by a maintainer to run workflows.

**Comment by RobertTheBlair** [comment]
Just fixed number of associated issue https://github.com/google/error-prone/issues/3396

**Review by cushon — APPROVED** [review]
Thanks for the contribution!

**Comment by RobertTheBlair** [comment]
> Thanks for the contribution!
Happy to help.  Hardest part was getting corp sign-off (-;

Is there a person among the maintainers I should tag to unblock the workflow steps?

**Comment by RobertTheBlair** [comment]
CI failed for only JDK 20 on Ubuntu, in code untouched by the branch.  Fresh merge & rebase done.

**Comment by RobertTheBlair** [comment]
I should update the docs in /docs/bugpattern/DirectInvocationOnMock.md as well.

**Comment by RobertTheBlair** [comment]
> I should update the docs in /docs/bugpattern/DirectInvocationOnMock.md as well.

Done

**Pull request closed without merging**
